src/launch/automation/helm/functions.py

```diff
@@ -35,12 +36,13 @@ def resolve_dependencies(
                 f"Remembering dependency {dependency['name']} with version {dependency['version']}."
             )
         elif global_dependencies[dependency["name"]] != dependency["version"]:
-            logger.exception(
+            conflict_message = (
                 f"Dependency {dependency['name']} has conflicting versions: "
                 f"{global_dependencies[dependency['name']]} and {dependency['version']}. "
                 "You must resolve this conflict before continuing."
             )
-            sys.exit(1)
+            logger.exception(conflict_message)
+            RuntimeError(conflict_message)
```

Comment:
```suggestion
raise RuntimeError(conflict_message)
```

This not having a `raise` tells me the test doesn't fully exercise this branch or doesn't detect the raise of an exception, which we should change. See comment below in the test file for a suggestion on how to correctly detect errors raised by a function under test.